src/services/refactors/extractSymbol.ts

```diff
@@ -1009,7 +1009,9 @@ namespace ts.refactor.extractSymbol {
 
         // Make a unique name for the extracted variable
         const file = scope.getSourceFile();
-        const localNameText = getUniqueName(isClassLike(scope) ? "newProperty" : "newLocal", file);
+        const localNameText = isPropertyAccessExpression(node) && !isClassLike(scope) && !checker.resolveName(node.name.text, node, SymbolFlags.Value, /*excludeGlobals*/ false) && !isPrivateIdentifier(node.name) && !isKeyword(node.name.originalKeywordKind!)

```

Comment:
I know that this line is long… :joy:

I did not extract it into a function as my initial goal was to minimize the amount of structural change required.

…but to pave a way for more inference to take place, I did some refactoring and extract them to `getLocalNameText` function. If [these diffs](https://github.com/dtinth/TypeScript/compare/396bc2e31ca0c3b905a1d3dea746eb8e34bde7a2...aaf760a0345397c985e50468acbc8947025cf8e2) look good, I can cherry pick them into this PR.

---

…and with the above refactoring, I was able to make it also infer local variable name by the expression’s type [(diff)](https://github.com/dtinth/TypeScript/compare/aaf760a0345397c985e50468acbc8947025cf8e2...194411d1c4e97430403c6c0fb4bfa4d41271c826). If this sounds like a good idea, I can open an another issue and submit that as another PR. (I prefer to limit the scope of this PR to just PropertyAccess, but if you prefer I can put them into this PR as well.)

![Screen recording](https://user-images.githubusercontent.com/193136/79046453-445bc100-7c3b-11ea-8e20-494910a20c48.gif)
_Example of inferring variable name based on expression’s type (not included in this PR)_